### PR TITLE
Independent checkpointing and evaluation period

### PR DESCRIPTION
**Pull request by le1nux**
Allows to define the logging intervals in the config for 
* training loss and troughput
* evaluation 
* checkpointing

### CODE REVIEW

File: src/modalities/trainer.py
Comment (mali-git) [inline src/modalities/trainer.py:1]:
```
def _get_local_sample_id(batch_id: int, batch_size: int) -> int:
        return (batch_id + 1) * batch_size - 1
```

Why do we reduce the batch_size by 1?

File: src/modalities/util.py [hunk 1]
```diff
@@ -21,12 +21,10 @@ def parse_enum_by_name(name: str, enum_type: Type[Enum]) -> Enum:
 
 
 def get_callback_interval_in_batches_per_rank(
```
Comment (le1nux) [inline src/modalities/util.py:23]: unused method

**Review by mali-git — APPROVED** [review]
